### ISSUE TITLE
Resizing the window can result in an error being thrown if the map element is removed in the meantime

### ISSUE THREAD

**Issue by MarkFalconbridge** [Leaflet/Leaflet#6981]
This issue arises if the map element is removed just after the window is resized.

It results in a similar error as is thrown by the cases mentioned in this issue - https://github.com/Leaflet/Leaflet/issues/5774.

What you see is an error in the browser console:

DomUtil.js:247 Uncaught TypeError: Cannot read property '_leaflet_pos' of undefined
    at getPosition (DomUtil.js:247)
    at NewClass._getMapPanePos (Map.js:1475)
    at NewClass.containerPointToLayerPoint (Map.js:1049)
    at NewClass._getCenterLayerPoint (Map.js:1512)
    at NewClass.getCenter (Map.js:837)
    at NewClass.panInsideBounds (Map.js:503)
    at NewClass._panInsideMaxBounds (Map.js:1280)
    at NewClass.fire (Events.js:190)

See here for a repro case - https://plnkr.co/edit/XSHvj1jLmz6M81RT4zWu?p=preview. To reproduce, change the size of the window containing the map by moving the splitter between the code window and the preview window. You should see the error in the browser console.

The error is thrown because the Map.invalidateSize method uses setTimeout with a timeout of 200ms. Map.invalidateSize even stores the handle returned from the setTimeout call as this._sizeTimer. The fix would be to call clearTimeout(this._sizeTimer) from the Map.remove method.

**Comment by IvanSanchez** [comment]
@MarkFalconbridge Nice analysis! Sounds like you've got it covered - would you like to send a PR to fix this, please?

**Comment by MarkFalconbridge** [comment]
This has already been fixed by #6958. I was running against release 1.6.0 which doesn't have that fix.

**Comment by MarkFalconbridge** [comment]
On further investigation I don't think #6958 fixes this issue. Applying a patch that uses the code from #6958 still gives the same error.

**Comment by MarkFalconbridge** [comment]
Here (https://jsfiddle.net/dr7k9mp3/2/) is an updated example using jsfiddle that illustrates the problem. Once loaded resize the pane hosting the map and then check the browser console.

**Comment by MarkFalconbridge** [comment]
The issue is fixed by a combination of #6958 and #6867. Please make a release soon to address this issues and avoid duplicate issues being raised.

**Comment by t1a2l** [comment]
@stephent having the same issue, it is unpredictable - happens sometimes. can't reproduce. found any solution for that?

**Comment by stephent** [comment]
@t1a2l for me the issue occurred because my code was briefly loading a Leaflet map then switching to another map technology as soon as the user preferences were received.

I added logic to prevent anything loading until the user prefs were received and the problem has not recurred. It seems that it might have been related to Leaflet being loaded and then unloaded before a callback had completed (that sounds very vague, I know - that's because I can't pretend I actually understand what was going on entirely!)

**Comment by RZsam** [comment]
have the same issue.
any solution?

**Comment by studiobrain** [comment]
seeing a very similar error:
```
TypeError: Cannot read property '_leaflet_pos' of undefined
    at _t (DomUtil.js:247)
    at t._getMapPanePos (Map.js:1475)
    at t.containerPointToLayerPoint (Map.js:1049)
    at t._getCenterLayerPoint (Map.js:1512)
    at t.getCenter (Map.js:837)
    at t.flyTo (Map.js:367)
```

**Comment by alexsalajan** [comment]
Still having the same issue with version 1.7.1

**Comment by theseanco** [comment]
Having this issue also with 1.7.1

**Comment by lagst3r** [comment]
I am having the same issue but only after when I called map2.remove().

**Comment by ASBondarev** [comment]
Having same issue with react-leaflet, trying to getPixelBounds() out of map in useState

**Comment by yniz** [comment]
Same issue when the wrapping Angular component is removed from the page, not using map.remove or it creates a second error - another instance is trying to use the element.

**Comment by Zae** [comment]
Don't want to spam the thread, but I'm also still seeing this issue with 1.7.1

**Comment by johnd0e** [comment]
@Zae You are right, this thread is spammed with unrelated (though quite similar) reports.

>  still seeing this issue

What exact issue do you mean?

**Comment by Zae** [comment]
Hi @johnd0e,

I see there is another thread that is still open that refers to the same issue  (#5774), I thought this was the only one referring to the `_leaflet_pos` error I was getting, same for the people above me probably.

I guess there are multiple ways of getting the `Cannot read property '_leaflet_pos' of undefined` error and while this one was fixed, it is still possible to get the same error.

Do you want people to add different reproductions to #5774 if they encounter this error or open a new issue?

**Comment by johnd0e** [comment]
There is a very little sense to add notes to this closed issue, so I'm going lo lock it.

> Do you want people to add different reproductions to #5774 if they encounter this error or open a new issue?

If your conditions fall into described in #5774 description - add note there, otherwise open new issue.
